### PR TITLE
Lpd 15418

### PR DESCRIPTION
**Pull request by noornajj**
https://liferay.atlassian.net/browse/LPD-15418

### CODE REVIEW

**Comment by noornajj** [comment]
ci:test:sf

**Comment by noornajj** [comment]
ci:test:publications

**Comment by noornajj** [comment]
ci:test:publications

File: modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/indicator/test/ChangeTrackingIndicatorConflictIconTest.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+			JSONObject jsonObject = _getConflictInfoJSONObject(
+				mockLiferayResourceResponse);
+
+			Assert.assertEquals(
+				"change-tracking-conflict-icon-danger",
+				String.valueOf(jsonObject.get("conflictIconClass")));
+			Assert.assertEquals(
+				"warning-full",
+				String.valueOf(jsonObject.get("conflictIconName")));
+		}
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/indicator/test/ChangeTrackingIndicatorConflictIconTest.java:120]: I see these same steps of assertions in every test cases. Let's extract it to a helper method (i.e. `_assertGetConflictInfo(String expectedConflictIconClass, String expectedConflictIconName)` to reduce duplicate code.

File: modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/indicator/test/ChangeTrackingIndicatorConflictIconTest.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+			Assert.assertEquals(
+				"check", String.valueOf(jsonObject.get("conflictIconName")));
+		}
+	}
+
+	@Test
+	public void testConflictIconWithPossibleConflict() throws Exception {
+		try (SafeCloseable safeCloseable =
+				CTCollectionThreadLocal.setCTCollectionIdWithSafeCloseable(
+					_ctCollection2.getCtCollectionId())) {
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/indicator/test/ChangeTrackingIndicatorConflictIconTest.java:157]: If `_ctCollection2` is only needed for this test, let's declare it locally..

**Review by pei-jung — CHANGES_REQUESTED** [review]

**Comment by noornajj** [comment]
ci:test:sf

File: modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/portlet/action/GetConflictInfoMVCResourceCommand.java [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+					"conflictIconClass", "change-tracking-conflict-icon-danger"
+				).put(
+					"conflictIconLabel",
+					_language.get(
+						themeDisplay.getLocale(), "conflict-detected-help")
+				).put(
+					"conflictIconName", "warning-full"
+				);
+			}
+			else if (possibleConflictCollection != null) {
```
[the remark this answers is not in the document]
Comment (noornajj) [reply modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/portlet/action/GetConflictInfoMVCResourceCommand.java]: Hi @pei-jung 
My understanding here is to change the else to an if statement, correct? If not can you please clarify?

**Comment by noornajj** [comment]
Hi @pei-jung 
Please see new changes

**Comment by pei-jung** [comment]
See https://github.com/liferay-publications/liferay-portal/pull/1040

**Pull request closed without merging**